types/session_cost.yaml

```diff
@@ -1,5 +1,64 @@
 description: Types to provide the session price
 types:
+  IdlePrice:
+    description: >-
+      Price components when idle
+    type: object
+    additionalProperties: false
+    properties:
+      grace_minutes:
+        description: >-
+          Grace period in minutes before idle time is charged.
+          Grace minutes start counting from the timestamp of the message in which state
+          changed from "Charging" to "Idle".
+        type: integer
+      hour_price:
+        description: >-
+          Price per hour while idle
+        type: number
+  CostCategory:
+    description: Cost category
+    type: string
+    enum:
+      - Energy
+      - Time
+      - FlatFee
+      - Other
+  ChargingPriceComponent:
+    description: >-
+      Charging price for a category
+    type: object
+    properties:
+      category:
+        description: >-
+          Component type for this charging price
+        type: string
+        $ref: /session_cost#/CostCategory
+      price:
+        description: >-
+          Price for this component
+        type: object
+        $ref: /money#/Price
```

Comment:
money is probably not the right thing here. The price.yaml defines the amount as integer in cents but it can be in this case that the kWh price is even a fraction of that. You sometimes see this for example at gas stations.

Reply:
What should it be then?
Price has a currency, which has a 'number of decimals' and currency. So with the 'number of decimals' you can define how many decimals the number has.